chalice/cli/__init__.py

```diff
@@ -13,20 +12,25 @@
 
 import botocore.exceptions
 import click
-from typing import Dict, Any  # noqa
+from botocore.session import Session  # noqa
+from typing import Dict, Any, Optional  # noqa
 
 from chalice import __version__ as chalice_version
 from chalice import prompts
 from chalice.app import Chalice  # noqa
 from chalice.awsclient import TypedAWSClient
-from chalice.cli.utils import create_botocore_session
-from chalice.config import Config
-from chalice.deploy import deployer
+from chalice.cli.factory import CLIFactory
+from chalice.config import Config  # noqa
 from chalice.logs import LogRetriever
-from chalice.package import create_app_packager
 from chalice.utils import create_zip_file, record_deployed_values
 
 
+# This is the version that's written to the config file
+# on a `chalice new-project`.  It's also how chalice is able
+# to know when to warn you when changing behavior is introduced.
+CONFIG_VERSION = '2.0'
```

Comment:
We should maybe throw errors if we encounter a config version higher than that version of the chalice can support